src/food/ood/loss.py

```diff
@@ -0,0 +1,55 @@
+import torch
+import torch.nn.functional as F
+
+
+class glass_reg(torch.nn.Module):
+    def __init__(self):
+        super(glass_reg, self).__init__()
+
+    def forward(self, model: torch.nn.Module, V: torch.nn.Module, target: torch.Tensor) -> torch.Tensor: #embedding - i-th col - phi(x_i), label_matrix - V, target - i-th col - i-th labels
+
+        n_classes = V.weight.shape[0]
+        target_sparse = torch.sparse.torch.eye(n_classes).to(target.device)
+        target_sparse = target_sparse.index_select(0, target)
+
+        A = torch.matmul(target_sparse.T,target_sparse)
+        diag = torch.diag(A)
+        ind = diag.nonzero(as_tuple = True)
+        diag[ind] = 1./ diag[ind]
+        Z = torch.diag(diag)
+
+        reg = torch.matmul(V.weight,V.weight.T) - (torch.matmul(A,Z)+torch.matmul(Z,A))/2
+        L=torch.norm(reg,p='fro')/n_classes**2
```

Comment:
Поправь ВЕЗДЕ по pep8. Код не особо читаемый. Можно воспользоваться autopep8. Я внесу этот момент в CI.
```suggestion
        L = torch.norm(reg,p='fro' ) / n_classes ** 2
```